routers/user/oauth.go

```diff
@@ -370,6 +371,42 @@ func AccessTokenOAuth(ctx *context.Context, form auth.AccessTokenForm) {
 	}
 }
 
+type OauthUserInfo struct {
+	Sub               string `json:"sub"`
+	Name              string `json:"name"`
+	PreferredUsername string `json:"preferred_username"`
+	Profile           string `json:"profile"`
+	Picture           string `json:"picture"`
+	Email             string `json:"email"`
+	Language          string `json:"language"`
+	Locale            string `json:"locale"`
+}
+
+// GetUserinfoOauth return json userinfo as in spec:
+func GetUserinfoOauth(ctx *context.Context) {
+	user := convert.ToUser(ctx.User, ctx.IsSigned, ctx.User != nil)
+	app_url := setting.AppURL
+	if setting.AppSubURL != "" {
```

Comment:
If there exist a better way to get the Profile url just tell me ...

Reply:
`user.HomeLink()`

Reply:
@adelowo nice to know! thx ... but it looks like @lafriks is working on this already ... :=